ethereum-optimism/ecosystem-contributions
# C-ATTS: An Attestation Data Duplicator

Hi, I am building C–ATTS, a platform to create, transform, combine and migrate EAS attestations. 

A C–ATTS attestation is defined by a “recipe”. A recipe is a combination of a GraphQL query, a processing algorithm and an output schema. The output of running a recipe is a regular EAS attestation that is 100% compatible with the EAS standard.

C–ATTS supports queries to the EAS API as well as queries to any subgraph hosted on The Graph. This means that C–ATTS can attest to virtually any data stored on any EVM chain.

Some example recipes could be:
- Migrate my Gitcoin Passport score from chain A to chain B
- Calculate my total value locked in DeFi on chain A
- Aggregate my NFT holdings on chain A
- Attest to my membership in a DAO 
- Attest to me owning a specific POAP
- Attest to me being a Gitcoin Grant donor

A C–ATTS recipe can include multiple queries, spanning multiple chains. The processing algorithm can be as simple as summing up the results of the queries or as complex as you can imagine. 

Once a recipe is setup, users can claim attestations. Think of it a bit like POAPs. 

Currently, C–ATTS is looking for projects that want to integrate composite attestations into their platform. If you are interested, please reach out to me at [Twitter](https://twitter.com/kristoferlund) or [Telegram](https://t.me/kristoferkristofer).

You can find more information and try the prototype app at https://catts.run

<img width="1195" alt="image" src="https://github.com/ethereum-optimism/ecosystem-contributions/assets/9698363/153a044d-91d9-4c0d-a5b6-77fbd0279c49">

The app is still in early development and I am looking for feedback and feature requests. Curretntly, you as a user can claim attestations from a number of pre-defined recipes. On the roadmap is a number of planned features:
- Create your own recipes
- Verify the processing logic of a recipe
- SDK for easy integration into your project

🙏 Thanks!
